### ISSUE TITLE
[TRACKER] Roll PyTorch action

### ISSUE THREAD

**Issue by vivekkhandelwal1** [llvm/torch-mlir#1690]
I'm creating this issue so that we can keep track of the errors that we see while updating the PyTorch version for the project. Anyone who faces an issue can post it here, try to fix that or if they don't know how to do that, can ask someone (working on that stuff) to fix the issue. I hope this approach looks fine.

cc: @powderluv @silvasean @ramiro050 @tanyokwok @eric-k256

**Comment by vivekkhandelwal1** [comment]
Hi @silvasean, the issue right now is with the TorchDynamo tests (https://github.com/llvm/torch-mlir/actions/runs/3639158147/jobs/6142206274). Could you take a look at this?

**Comment by silvasean** [comment]
For such failures we should XFAIL the relevant tests and file a bug like this for each different error message: https://github.com/pytorch/pytorch/issues/89630

You should be able to just take the "forward" method of the e2e test and paste it into a .py file like in the example issue I showed.

As described in the [long-term roadmap](https://github.com/llvm/torch-mlir/blob/main/docs/long_term_roadmap.md#refactoring-the-frontend) the Torch-MLIR project's role here is to be a "good citizen power user" of the new PyTorch infra.

```
These blockers are expected to be addressed by the PyTorch core devs over time. Torch-MLIR's role here is to communicate our requirements to PyTorch core and align their roadmap and ours. We do this by maintaining connections with the PyTorch core developers and being "good-citizen power users" of their latest technology (i.e. trying things out, surfacing bugs, providing feedback, etc.).
```


Do we need to start a RollPyTorch rotation @powderluv @ashay ? What is our current approach here?

**Comment by powderluv** [comment]
Typically nod.ai team did the Roll PyTorch manually. With @ashay 's help we were able to automate it and we usually fixed up issues except if it was LTC which was going through stabilization and we would defer to the Cerebras team.  We nod.ai team can still continue to do that and be first line to bump PyTorch if @silvasean or @ramiro050 can help with TorchDynamo failures (just like Cerebras folks handled LTC). In the long run once torchdynamo stabilizes we can also take over that part or get others in a rotation. Right now outside of @ramiro050 / @silvasean the knowledge of TorchDynamo is minimal.

**Comment by ashay** [comment]
I agree with @powderluv.  The resolution for PyTorch update failures has been largely dependent on the kind of failure -- whoever was most familiar with the failures resolved them, so I don't think a rotation would be helpful.  That said, until Vivek created this issue, there wasn't a good place to call attention to PyTorch update problems, so going forward, I imagine it should be easier to get the right folks' attention to triage and fix issues.

**Comment by silvasean** [comment]
Happy to help with issues with TorchDynamo. In this case the issues are very mechanical to repro and don't really touch on anything TorchDynamo-specific, so I would advise to file upstream bugs as I described and add them to the XFAIL.

**Comment by vivekkhandelwal1** [comment]
> Happy to help with issues with TorchDynamo. In this case the issues are very mechanical to repro and don't really touch on anything TorchDynamo-specific, so I would advise to file upstream bugs as I described and add them to the XFAIL.

Okay, I will add them to xfail and create an issue in PyTorch. There's another failure with the latest torch version, I will address that as well in the same PR.

**Comment by ashay** [comment]
The RollPyTorch action is failing because of what seems like an error in compiling the LTC code:

```
/main_checkout/torch-mlir/python/torch_mlir/csrc/base_lazy_backend/mlir_native_functions.cpp:312:46: error: too few arguments to function call, expected 3, have 1
  auto size = c10::asIntArrayRefSlow(sym_size);
```

@antoniojkim Would you be able to resolve it please? Thanks!

https://github.com/llvm/torch-mlir/actions/runs/3704365156/jobs/6276901149

**Comment by vivekkhandelwal1** [comment]
> The RollPyTorch action has failed. See [CI log](https://github.com/llvm/torch-mlir/actions/runs/4174092627) for details.
> 
> The following information may come handy when fixing the code.
> 
> ```
> torch version: 2.0.0.dev20230214
> torch commit hash: e67dc17b067c70fd238da44598c698cfa97581f6
> torchvision version: 0.15.0.dev20230214
> ```

Hi, after these changes in PyTorch (https://github.com/pytorch/pytorch/commit/ceb0f1576b201adf8cafbde32a9ed7ba95ead025), a lot of tests are failing in Torch-MLIR for Torchdynamo config. I have added an op (https://github.com/llvm/torch-mlir/pull/1882) to fix some of the tests. But even after this, there are nearly 20 tests that are failing some with runtime errors, and others with the values mismatch. We need to fix them in order to get the roll PyTorch working. Any help in this would be appreciated.

cc: @silvasean @ramiro050

**Comment by ashay** [comment]
The above error seems to be happening because `unzip` isn't available on the new runner images.  PR https://github.com/llvm/torch-mlir/pull/1893 fixes the problem.

**Comment by vivekkhandelwal1** [comment]
> The RollPyTorch action has failed. See [CI log](https://github.com/llvm/torch-mlir/actions/runs/4397596876) for details.
> 
> The following information may come handy when fixing the code.
> 
> ```
> torch version: 2.1.0.dev20230312
> torch commit hash: 1238ae3e3954f69a5de91b8ee04b718301b7f193
> torchvision version: 0.15.0.dev20230312
> ```

This is an LTC-related error probably generated because of this commit https://github.com/pytorch/pytorch/commit/d0e4ca233ec21a75a3f1365b0c0c4480cc7de1d3. @antoniojkim @ke1337 could you please take a look at this issue?

**Comment by antoniojkim** [comment]
Raised an issue in PyTorch:
https://github.com/pytorch/pytorch/issues/96676

**Comment by ashay** [comment]
@vivekkhandelwal1 The only discrepancy that I can think of is that we build the pytorch WHL file in CI whereas you may (?) have used pre-build WHL files using pip when trying to reproduce the problem.

**Comment by vivekkhandelwal1** [comment]
> > The RollPyTorch action has failed. See [CI log](https://github.com/llvm/torch-mlir/actions/runs/4543405704) for details.
> > The following information may come handy when fixing the code.
> > ```
> > torch version: 2.1.0.dev20230328
> > torch commit hash: 81558034b3e4c3cbbdd13cb52134d69f19905820
> > torchvision version: 0.16.0.dev20230328
> > ```
> 
> Hi, I think this [issue](https://github.com/llvm/torch-mlir/actions/runs/4543405704) is happening over the CI only. Because when I ran the tests locally they ran successfully for the linalg and torchdynamo config for me. cc: @ashay @silvasean @ramiro050

I have created an issue in Torchvision regarding this error. If anyone has any inputs regarding this issue then please post it over here: https://github.com/pytorch/vision/issues/7472.

**Comment by vivekkhandelwal1** [comment]
The Roll-PyTorch CI in Torch-MLIR does run but it doesn't update the torch version. It has been happening for quite a few days. The CI for committing the changes is somehow skipped. https://github.com/llvm/torch-mlir/actions/runs/6112908899

CC: @ashay @stellaraccident @powderluv

**Comment by ashay** [comment]
Dang, sorry.  Will take a look at this soon.  Thanks for letting me know that it was broken.

**Comment by powderluv** [comment]
Potential fix https://github.com/llvm/torch-mlir/pull/2446 probably happened when we switched keys from Sean to Stella.
